File: cpp/Lec 2 Build-up Logical Thinking (Patterns)/pattern_14.cpp
// A
// AB
// ABC
// ABCD
// ABCDE
// BRUTE FORCE
#include <iostream>
using namespace std;
int main()
{
    int n = 5;
    for (int i = 1; i <= n; i++)
    {
        for (char c = 'A'; c < 'A' + i; c++)
        {
            cout << c;
        }
        cout << endl;
    }
    return 0;
}

// BETTER
#include <iostream>
#include <string>
using namespace std;
int main()
{
    int n = 5;
    for (int i = 1; i <= n; i++)
    {
        cout << string(i, '0') << endl;
    }
    return 0;
}

// BEST
#include <iostream>
#include <string>
using namespace std;
int main()
{
    int n = 5;
    string s;
    for (char c = 'A'; c < 'A' + n; c++)
    {
        s += c;
        cout << s << '\n';
    }
    return 0;
}

File: python/Lec 2 Build-up Logical Thinking (Patterns)/pattern_14.py
# A
# AB
# ABC
# ABCD
# ABCDE

# BRUTE FORCE
n = 5
for i in range(1, n + 1):
    line = ""
    for j in range(i):
        line += chr(ord('A') + j)
    print(line)

# BETTER
n = 5
for i in range(1, n + 1):
    print(''.join(chr(ord('A') + j) for j in range(i)))

# BEST
n = 5
s = ""
for i in range(n):
    s += chr(ord('A') + i)
    print(s)